src/guest_memory.rs

```diff
@@ -246,6 +246,17 @@ pub trait GuestMemoryRegion: Bytes<MemoryRegionAddress, E = Error> {
     unsafe fn as_mut_slice(&self) -> Option<&mut [u8]> {
         None
     }
+
+    /// Returns a [`VolatileSlice`](struct.VolatileSlice.html) of `count` bytes starting at
+    /// `offset`.
+    fn get_slice(
```

Comment:
The default implementation could return HostAddressNotAvailable, like get_host_address.

Reply:
There is a problem that way, `get_slice` needs to return `volatile_memory::Result`, while `HostAddressNotAvailable` doesn't match. We can return something like `Err(volatile_memory::Error::OutOfBounds { addr: offset })` instead, but the meaning is not precise. 

So, how about keeping it without default implementation?

Reply:
I think you should not return `volatile_memory::Result` but `guest_memory::Result`; I missed it in my original review. There is already a `From<>` implementation, so you can just write

```rust
    let slice = self.mapping.get_slice(offset, count)?
    Ok(slice)
```

and the `?` operator will take care of the conversion (unfortuantely you cannot just write `self.mapping.get_slice(offset, count).into()`, I think).

Reply:
Thank you for detailed explanation and the example. Implementing `From` make the Result handling among different traits easy and graceful!
I've updated the code.